posthog/queries/funnels/funnel_event_query.py

```diff
@@ -112,17 +86,35 @@ def get_query(self, entities=None, entity_name="events", skip_entity_filter=Fals
 
         null_person_filter = f"AND notEmpty({self.EVENT_TABLE_ALIAS}.person_id)" if self._using_person_on_events else ""
 
-        query = f"""
-            SELECT {', '.join(_fields)} FROM events {self.EVENT_TABLE_ALIAS}
+        select_fragment = (
+            f"SELECT {', '.join(_fields)}" + "{extra_select_fields}" + f" FROM events {self.EVENT_TABLE_ALIAS}"
+        )
+        joins_fragment = (
+            f"""
             {self._get_distinct_id_query()}
             {person_query}
             {groups_query}
+        """
+            + "{extra_join}"
+        )
+        where_fragment = (
+            f"""
             WHERE team_id = %(team_id)s
             {entity_query}
             {date_query}
             {prop_query}
             {null_person_filter}
         """
+            + "{step_filter}"
+        )
+
+        query = f"""
```

Comment:
Q: Why split the overall thing up? IMO not adding too much clarity here.

Reply:
We spoke about this on a call - essentially there are ordering requirements with the functions building this query (as they have side effects) and this was the cleanest way to build things now without a large refactor